packages/components/src/internal/components/samples/SampleSetSummary.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { FC, memo, ReactNode, useCallback, useEffect, useState, useMemo } from 'react';
+
+import { Filter } from '@labkey/api';
+
+import { GridPanelWithModel, SCHEMAS, AppURL, SelectInput, User, Location } from '../../..';
+
+import { SampleSetCards } from './SampleSetCards';
+import { SampleSetHeatMap } from './SampleSetHeatMap';
+
+const SELECTION_HEATMAP = 'heatmap';
+const SELECTION_CARDS = 'cards';
+const SELECTION_GRID = 'grid';
+
+const SAMPLESET_VIEW_OPTIONS = [
+    { value: SELECTION_CARDS, label: 'Cards' },
+    { value: SELECTION_GRID, label: 'Grid' },
+    { value: SELECTION_HEATMAP, label: 'Heatmap' },
+];
+
+const SAMPLE_SET_GRID_GRID_ID = 'samplesets-grid-panel';
+
+const SAMPLE_QUERY_CONFIG = {
+    urlPrefix: 'samplesetgrid',
+    isPaged: true,
+    id: SAMPLE_SET_GRID_GRID_ID,
+    schemaQuery: SCHEMAS.EXP_TABLES.SAMPLE_SETS,
+};
+
+interface SampleSetSummaryProps {
+    location?: Location;
+    navigate: (url: string | AppURL) => any;
+    user: User;
+    excludedSampleSets?: string[];
+}
+
+export const SampleSetSummary: FC<SampleSetSummaryProps> = memo(props => {
+    const { location, navigate, user, excludedSampleSets } = props;
+
+    const [selected, setSelected] = useState<string>();
+
+    const queryConfig = useMemo(() => {
+        return {
+            ...SAMPLE_QUERY_CONFIG,
+            baseFilters: excludedSampleSets
+                ? [Filter.create('Name', excludedSampleSets, Filter.Types.NOT_IN)]
+                : undefined,
+        };
+    }, [excludedSampleSets]);
+
+    useEffect(() => {
+        setSelected(location?.query?.viewAs ?? 'grid');
+    }, [location]);
+
+    const showHeatMap = (): boolean => {
+        return selected === SELECTION_HEATMAP;
+    };
+
+    const showCards = (): boolean => {
+        return selected === SELECTION_CARDS;
+    };
+
+    const showGrid = (): boolean => {
+        return selected === SELECTION_GRID || selected === undefined;
+    };
+
+    const onSelectionChange = useCallback((selected, value) => {
+        setSelected(value);
+    }, []);
+
+    const renderSelect = (): ReactNode => {
```

Comment:
nit: `renderSelect` is not providing any value. Can just render the `<SelectInput>` inline.